notebooks/config.ipynb

```diff
@@ -0,0 +1,1046 @@
+    "## Realizing Configs to a Dictionary\n",
+    "\n",
+    "The `config.realize_to_dict()` function has the ability to manipulate config values, and returns the config as a Python `dict`.\n",
+    "<!--cell 50-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "id": "f6aa96d5-cdcd-4a41-91e2-76e1b1f1356b",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Help on function realize_to_dict in module uwtools.api.config:\n",
+      "\n",
+      "realize_to_dict(input_config: Union[uwtools.config.formats.base.Config, pathlib.Path, dict, str, NoneType] = None, input_format: Optional[str] = None, update_config: Union[uwtools.config.formats.base.Config, pathlib.Path, dict, str, NoneType] = None, update_format: Optional[str] = None, key_path: Optional[list[Union[str, int]]] = None, values_needed: bool = False, total: bool = False, dry_run: bool = False, stdin_ok: bool = False) -> dict\n",
+      "    Realize a config to a ``dict``, based on a base input config and an optional update config.\n",
+      "\n",
+      "    See ``realize()`` for details on arguments, etc.\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "help(config.realize_to_dict)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e44a3aa1-e7d5-4547-af22-541be10d65d8",
+   "metadata": {},
+   "source": [
+    "Like `realize()`, input or update configs can be Python dictionaries, UW `Config` objects, or files like the one below.\n",
+    "<!--cell 52-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 27,
+   "id": "bdc98ce1-e213-41ac-b1f2-03bd52238e30",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "greeting: Hello\n",
+      "recipient: World\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "cat fixtures/config/get-config.yaml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "eeacc37f-b808-4d0e-9e29-ff8aa1fc2ff2",
+   "metadata": {},
+   "source": [
+    "`realize_to_dict()` has the same parameters as `realize()`, with the exception of `output_file` and `output_format`. These parameters are not used because a config won't be written to a file or to `stdout` as `config.realize()` would. Instead, configs can be manipulated or converted to a `dict` without the need to specify an output file or format.\n",
+    "<!--cell 54-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "id": "ae7f648e-6586-4700-87e7-492ca3a02a06",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'id': '456', 'greeting': 'Hello', 'recipient': 'World'}\n"
+     ]
+    }
+   ],
+   "source": [
+    "config_dict = config.realize_to_dict(\n",
+    "    input_config={\"id\": \"456\"},\n",
+    "    update_config=\"fixtures/config/get-config.yaml\"\n",
+    ")\n",
+    "print(config_dict)"
```

Comment:
```suggestion
    "config.realize_to_dict(\n",
    "    input_config={\"id\": \"456\"},\n",
    "    update_config=\"fixtures/config/get-config.yaml\"\n",
    ")\n"
```
Since Jupyter displays the cells result anyway, this more concise version shows exactly the same output.